src/CMakeLists.txt

```diff
@@ -206,6 +207,8 @@ endif()
 if(UNIX AND NOT APPLE)
```

Comment:
Can you wrap your new code in a new `if(LINUX)` under this?
Otherwise you'll also enable it for all BSDs on which it probably wasn't tested so far.